### PR TITLE
fix: DBTP-972 Ignored by Checkov baseline

### PR DESCRIPTION
**Pull request by mramplin**

### CODE REVIEW

File: application-load-balancer/main.tf [hunk 1]
```diff
@@ -45,11 +52,12 @@ resource "aws_lb_listener" "alb-listener" {
...
 resource "aws_security_group" "alb-security-group" {
-  # checkov:skip=CKV2_AWS_5:Security group is used by VPC. Ticket to investigate: https://uktrade.atlassian.net/browse/DBTP-1039
-  for_each = local.protocols
-  name     = "${var.application}-${var.environment}-alb-${each.key}"
-  vpc_id   = data.aws_vpc.vpc.id
-  tags     = local.tags
+  # checkov:skip=CKV2_AWS_5: False Positive in Checkov - https://github.com/bridgecrewio/checkov/issues/3010
+  for_each    = local.protocols
+  name        = "${var.application}-${var.environment}-alb-${each.key}"
+  description = "Managed by Terraform"
```
Comment (gabelton) [inline application-load-balancer/main.tf:58]: Don't we already have a tag that tells us that this resource is managed by terraform?

```suggestion
  description = "Security group for application load balancer"
```

File: domain/main.tf [hunk 2]
```diff
@@ -3,11 +3,12 @@ data "aws_route53_zone" "root-zone" {
 }
 
 resource "aws_route53_zone" "new-zone" {
+  # checkov:skip=CKV2_AWS_39:  Requires wider discussion around log/event ingestion before implementing. To be picked up on conclusion of DBTP-974
+  # checkov:skip=CKV2_AWS_38:
```
Comment (gabelton) [inline domain/main.tf:7]: What's the reason for skipping this one? Same as above?

File: domain/main.tf [hunk 3]
```diff
@@ -3,11 +3,12 @@ data "aws_route53_zone" "root-zone" {
 }
 
 resource "aws_route53_zone" "new-zone" {
+  # checkov:skip=CKV2_AWS_39:  Requires wider discussion around log/event ingestion before implementing. To be picked up on conclusion of DBTP-974
+  # checkov:skip=CKV2_AWS_38:  
   for_each = toset(var.zones)
   name     = "${each.key}.${data.aws_route53_zone.root-zone.name}"
   tags     = local.tags
 }
-
```
Comment (gabelton) [inline domain/main.tf:10]: Do we want to remove this line break?

File: elasticache-redis/main.tf [hunk 4]
```diff
@@ -14,9 +14,12 @@ data "aws_subnets" "private-subnets" {
 }
 
 data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
 
 
 resource "aws_elasticache_replication_group" "redis" {
+  # checkov:skip=CKV_AWS_31:Potential cascading impact on Celery. Requires further analysis
+  # checkov:skip=CKV_AWS_191:Potential cascading impact on Celery. Requires further analysis
```
Comment (gabelton) [inline elasticache-redis/main.tf:22]: Should we create a spike / investigation ticket?

File: elasticache-redis/main.tf [hunk 5]
```diff
@@ -57,12 +60,14 @@ resource "aws_security_group" "redis" {
...
 
   ingress {
+    description = "Allow from VPC on port 6379"
     from_port   = 6379
     to_port     = 6379
     protocol    = "tcp"
     cidr_blocks = [data.aws_vpc.vpc.cidr_block]
   }
   egress {
+    description = "Allow all traffic"
```
Comment (gabelton) [inline elasticache-redis/main.tf:70]: Not really related to your PR, but does anyone remember why we needed to allow all traffic?

File: extensions/main.tf [hunk 6]
```diff
@@ -90,6 +90,8 @@ module "monitoring" {
 }
 
 resource "aws_ssm_parameter" "addons" {
+  # checkov:skip=CKV_AWS_337: Used by copilot potential changes to platform-helper before CMK can be enabled
```
Comment (gabelton) [inline extensions/main.tf:93]: I don't quite understand what this one means

File: postgres/rds.tf [hunk 7]
```diff
@@ -118,4 +136,4 @@ data "aws_iam_policy_document" "enhanced-monitoring" {
       identifiers = ["monitoring.rds.amazonaws.com"]
     }
   }
-}
+}
```
Comment (gabelton) [inline postgres/rds.tf:139]: github complaining about lack of newline
```suggestion
}

```

File: postgres/secrets.tf [hunk 8]
```diff
@@ -1,4 +1,5 @@
 resource "aws_ssm_parameter" "master-secret-arn" {
+  # checkov:skip=CKV_AWS_337: Need to determine downstream impact before moving to CMK
```
Comment (gabelton) [inline postgres/secrets.tf:2]: Should we create a ticket?

**Review by gabelton — COMMENTED** [review]
Satisfying to see all the deletions from `.checkov.baseline`! Left a few small comments, mostly around creating tickets for follow up work

File: domain/main.tf (hunk 3 above)
Comment (mramplin) [reply domain/main.tf:10]: Nope. Will add back in!

File: extensions/main.tf (hunk 6 above)
Comment (mramplin) [reply extensions/main.tf:93]: This one is an extension/learning from DBTP-1128. The ECS container has issues decrypting items in KMS that are encrypted by a CMK without changes to the commandline used by copilot. As the commandline is defined in platform-helper that means we need to update it to change the command line. Hope that makes more sense!

It's another one we should probably look to raise it's own ticket for

File: domain/main.tf (hunk 2 above)
Comment (mramplin) [reply domain/main.tf:7]: Looks like I just missed adding in the comment for this one. Will add in. This one is probably another one we need an investigation/spike ticket for.

File: extensions/main.tf (hunk 6 above)
Comment (mramplin) [reply extensions/main.tf]: Changed comment to hopefully clarify

Comment (gabelton) [reply extensions/main.tf]: Makes sense and agree it sounds like a separate ticket

File: application-load-balancer/main.tf (hunk 1 above)
Comment (mramplin) [reply application-load-balancer/main.tf:58]: I'm going to revert this one as changing the description makes it want to recreate the security groups.

File: domain/main.tf (hunk 2 above)
Comment (mramplin) [reply domain/main.tf]: JIRA raised and number added to comment

File: elasticache-redis/main.tf (hunk 4 above)
Comment (mramplin) [reply elasticache-redis/main.tf]: JIRA raised and number added to comment

File: postgres/secrets.tf (hunk 8 above)
Comment (mramplin) [reply postgres/secrets.tf]: JIRA raised and number added to comment

File: statefile-backend/main.tf [hunk 9]
```diff
@@ -80,4 +108,4 @@ resource "aws_dynamodb_table" "terraform-state" {
       purpose = "Terraform statefile lock - DBT Platform"
     }
   )
-}
+}
```
Comment (gabelton) [inline statefile-backend/main.tf:111]:
```suggestion
}

```

**Review by gabelton — APPROVED** [review]
One small newline suggestion, but approving now. Nice one!

**Pull request merged**